### DISCUSSION TITLE
Possible to refer to record type in same record definition?

### DISCUSSION THREAD

**Discussion by ewmailing** [pallene-lang/pallene#513] [Q&A] [answered]
I want to use the record type that I'm trying to define, sort of like how one would refer to itself in a linked list node struct definition in C.

I want to do something like:
```
record Foo
    next:Foo
end
```
and
```
record Bar
    listOfOtherBars:{Bar}
end
```

The compiler doesn't like this: 
```type 'Foo' is not declared compilation aborted due to previous error```

The compiler seems to let me workaround this with ```any```. 
First, I want to make sure that the any workaround is actually allowed and won't break other assumptions.
Second, I wanted to know if there is a way to explicitly do this without the use of ```any```.

**Comment by hugomg** [comment, marked as answer]
This is a very embarassing thing we haven't implemented yet: #85
That workaround with `any` is the best thing we can do for now.
